demo-project/ruff.toml

```diff
@@ -0,0 +1,5 @@
+extend = "../ruff.toml"
+
+[lint.isort]
```

Comment:
Is this a best practice to have ruff.toml ? Can we do this in demo-project/pyproject.toml -> table [tool.ruff.lint.isort] ?

Reply:
`ruff` supports both `pyproject.toml` section and dedicated `ruff.toml` - see [here](https://docs.astral.sh/ruff/configuration/#config-file-discovery). So yes what you propose is feasible.

There is no explicit recommendation in `ruff` docs on which to prefer, and I have not heard of what's generally considered best practice. My thinking in general is that whenever possible, I'd prefer a dedicated config for every tool, because it:

1. Makes each config file have isolated responsibility
2. Simplifies version control a bit - e.g. 2 PRs where one changes dependencies and another changes the lint rulesets wouldn't touch the same file
3. Makes `pyproject.toml` smaller and more manageable

So I use `ruff.toml` in all of my projects, and never put that in `pyproject.toml` because of above.

Let me know if you disagree & still prefer the `pyproject.toml` - I can change to that. If so, then probably we'd need to configure `ruff` via `pyproject.toml` not only in `demo-project` but also in repo root and `package`.

Reply:
I came up with a better argument :)

1. It's useful to have `ruff` configuration **in repo root**, so that both `package`, `demo-project` or other **packages** can inherit it
2. Repo root is not a Python package, so `ruff.toml` is preferred over `pyproject.toml` in root
3. (1 + 2) sort of require us to use `ruff.toml` in the root, so for consistency, I suggest using `ruff.toml` files in `package` and `demo-project` too.

Reply:
Sure, I am happy to have ruff.toml, I wanted to understand if this was a better practice suggested from ruff. Thanks for the PR